File: php/phpfind/src/phpfind/FindOption.php
<?php

declare(strict_types=1);

namespace phpfind;

/**
 * Class FindOption
 */
readonly class FindOption
{
    public string $short_arg;
    public string $long_arg;
    public string $desc;
    public ArgTokenType $arg_type;
    public string $sort_arg;

    /**
     * @param string $short_arg
     * @param string $long_arg
     * @param string $desc
     * @param ArgTokenType $arg_type
     */
    public function __construct(string $short_arg, string $long_arg, string $desc, ArgTokenType $arg_type)
    {
        $this->short_arg = $short_arg;
        $this->long_arg = $long_arg;
        $this->desc = $desc;
        $this->arg_type = $arg_type;
        $this->sort_arg = $this->__sort_arg();
    }

    /**
     * @return string
     */
    private function __sort_arg(): string
    {
        if ($this->short_arg) {
            return strtolower($this->short_arg) . 'a' . $this->long_arg;
        }
        return $this->long_arg;
    }
}

File: objc/objcfind/Sources/objcfind/FindOption.m
#import "FindOption.h"

@implementation FindOption

- (instancetype) initWithShortArg:(NSString *)sArg withLongArg:(NSString *)lArg withDesc:(NSString *)desc withArgType:(ArgTokenType)argType {
    self = [super init];
    if (self) {
        self.shortArg = sArg;
        self.longArg = lArg;
        self.desc = desc;
        self.argType = argType;
    }
    return self;
}

-(NSString *) sortArg {
    if (self.shortArg) {
        return [NSString stringWithFormat:@"%@@%@", [self.shortArg lowercaseString], [self.longArg lowercaseString]];
    }
    return [self.longArg lowercaseString];
}

- (NSString *) description {
    NSMutableString *d = [[NSMutableString alloc] initWithString:@"FindOption("];
    [d appendFormat:@"short: \"%@\", long: \"%@\", desc: \"%@\")",
     self.shortArg, self.longArg, self.description];
    return d;
}

@end
